server/server.go

```diff
@@ -724,6 +724,10 @@ func (s *Server) addInternalRoutes(entryPointName string, router *mux.Router) {
 	if s.globalConfiguration.API != nil && s.globalConfiguration.API.EntryPoint == entryPointName {
 		s.globalConfiguration.API.AddRoutes(router)
 	}
+
+	if s.leadership != nil {
```

Comment:
You can't put this route on all entrypoints.
This route must be activated only on the entrypoint where we activate `--api`. So you need to put it in the API `if`.

Moreover, I think the endpoint should be /api/cluster/leader, WDYT?